lib/Value_Type.thy

```diff
@@ -41,6 +41,8 @@ fun force_nat_numeral (Const (@{const_name numeral}, Type ("fun", [num, _])) $ n
   | force_nat_numeral (Const (@{const_name "Groups.zero"}, _)) = @{term "0::nat"}
   | force_nat_numeral t = raise TERM ("force_nat_numeral: number expected", [t])
 
+fun cast_to_nat t = if type_of t = @{typ int} then @{term nat} $ t else t
```

Comment:
This has an interesting implication for negative numbers. What happens there? Maybe we should have the comment above say "a positive numeral"?

Reply:
I don't think a negative number is a numeral (it's unary minus applied to a numeral), so it'll likely just throw an exception in the code that forces it to be a numeral.

Reply:
Have added "positive" for more clarity.